spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/ExtendedBindingProperties.java

```diff
@@ -22,10 +22,38 @@
  *
  * @author Marius Bogoevici
  * @author Mark Fisher
+ * @author Soby Chacko
  */
 public interface ExtendedBindingProperties<C, P> {
 
 	C getExtendedConsumerProperties(String channelName);
 
 	P getExtendedProducerProperties(String channelName);
+
+	/**
+	 * Extended binding properties can define a default prefix to place all the extended
+	 * common producer and consumer properties. For example, if the binder type is foo
+	 * it is convenient to specify common extended properties for the producer or consumer
+	 * across multiple bindings in the form of `spring.cloud.stream.foo.default.producer.x=y`
+	 * or `spring.cloud.stream.foo.default.consumer.x=y`.
+	 *
+	 * The binding process will use this defaults prefix to resolve any common extended
+	 * producer and consumer properties.
+	 *
+	 * @return default prefix for extended properties
+	 * @since 2.1.0
+	 */
+	String getDefaultsPrefix();
+
+	/**
+	 *
+	 * Extended properties class against which default extended producer and consumer properties
+	 * are resolved. It is expected that this class has two properties - one called producer
+	 * and another called consumer that contains the extended properties for producer and
+	 * consumer respectively.
+	 *
+	 * @return extended properties class that contains extended producer/consumer properties
+	 * @since 2.1.0
+	 */
+	Class<?> getExtendedPropertiesEntryClass();
```

Comment:
Perhaps better to use `Class<? extends SomeInterface>` and access the properties via getters?

Reply:
Actually, after discussing it with Soby, I am ok with how it is at the moment given that `MergableProperties` is already reflection heavy and that the whole thing happens on initialization, but most importantly introducing a new interface (i.e., `ProducerConsumerAware`) would require changes to all binders (i.e., `RabbitBindingProperties`)
That said, I do agree that from API design perspective we would need to address it at some point.

Reply:
But it has nothing to do with the Binder, its the `xxxBindingProperties` - I don't see a big burden of making, say, `RabbitBindingProperties` implement something like `ProducerConsumerBinderSpecificPropertiesProvider`.

We could still use reflection to invoke `getProducer()` - my objection is using the field name in the reflection, rather than a getter.

Reply:
Actually, with

```java
public interface BinderSpecificPropertiesProvider {

	Object getConsumer();

	Object getProducer();

}
```

I don't think you'd need to use reflection at all (since you are only looking for an `Object`, and it's ok for the implementation to narrow the type, since it's still `Object`.